jgiven-core/src/main/java/com/tngtech/jgiven/impl/StandaloneScenarioExecutor.java

```diff
@@ -190,6 +190,8 @@ private void gatherRules( Object stage ) {
 
     private <T> T update( T t ) throws Throwable {
         if( currentStage == t ) { // NOSONAR: reference comparison OK here
+            readScenarioState( currentStage );
```

Comment:
This will have a big performance overhead. This actually what the if statement was good for: to avoid the overhead of reading and updating the state.